daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -416,7 +418,11 @@ case class PartialTransaction(
           key = ec.contractKey,
         )
         val nodeId = ec.nodeId
-        copy(context = ec.parent.addChild(nodeId), nodes = nodes.updated(nodeId, exerciseNode))
+        copy(
+          context = ec.parent.addChild(nodeId),
+          nodes = nodes.updated(nodeId, exerciseNode),
+          nodeSeeds = ec.parent.nextChildrenSeed.fold(nodeSeeds)(s => nodeSeeds :+ (nodeId -> s)),
```

Comment:
This looks in the wrong order. `ec.parent.addChild` two lines above drops the current seed for the exercise. So when we update `nodeSeeds`, we get the seed already for the next child rather than the current exercise.

Reply:
Oh forget it, contexts are immutable, so the order does not matter.

Reply:
I moved up the call to `ec.parent.nextChildrenSeed`, hopefully it will be clearer.